services/wiki/wiki.go

```diff
@@ -358,3 +358,34 @@ func DeleteWiki(ctx context.Context, repo *repo_model.Repository) error {
 	system_model.RemoveAllWithNotice(ctx, "Delete repository wiki", repo.WikiPath())
 	return nil
 }
+
+func ChangeDefaultWikiBranch(ctx context.Context, repo *repo_model.Repository, newBranch string) error {
+	return db.WithTx(ctx, func(ctx context.Context) error {
```

Comment:
Do we need to validate `newBranch`?

Reply:
It seems that it is necessary. 
If user convert the default wiki branch into an invalid value before creating the first wiki, they will get 500 error when creating wiki.
![image](https://github.com/go-gitea/gitea/assets/18380374/34b3a320-2927-41ec-9ab0-4f92308d50bd)

Reply:
68be709e6139e90f9146aa29ca8a258b712957bd